vulnfeeds/cves/versions.go

```diff
@@ -52,35 +53,141 @@ type CPE struct {
 	Other      string
 }
 
-func extractGitHubCommit(link string) *FixCommit {
-	// Example: https://github.com/google/osv/commit/cd4e934d0527e5010e373e7fed54ef5daefba2f5
-	u, err := url.Parse(link)
+// Returns the base repository URL for supported repository hosts.
+func Repo(u string) (string, bool) {
+	parsedURL, err := url.Parse(u)
 	if err != nil {
-		return nil
+		log.Printf("%+v", err)
```

Comment:
We should just return the error instead of logging it here, and have callers handle it. Same for `Commit()`.

Reply:
> We should just return the error instead of logging it here, and have callers handle it. Same for `Commit()`.

If we do that, then the simplicity of having a boolean return value for things working correctly is lost.

The current implementation was inspired from [go/gotip/episodes/30](https://goto.google.com/gotip/episodes/30)

If you don't feel that this is appropriate here, I can refactor accordingly.

Reply:
That example lists a motivating factor as there being only a single possible sentinel error value, which isn't very useful. In this case, we have have at least two: one where the url.Parse failed, and the other is if it's unsupported. 

I also generally like returning errors, because they can propagate useful context up the stack, and leave it up to the caller to decide when the most appropriate time to log is, as opposed to here where we just log regardless. 

Anyway, not going to split any hairs over this minor thing and happy to just leave this as is  :) This is something we can easily refactor in the future if we need to.